### PR TITLE
Expose in builtins' members internal type & size (on top of offset) in extension_api.json

### LINKED ISSUE #64189: [GDExtension] Vector*/Vector*i members type in extension_api.json leads to an incorrect size
### Godot version

4.0-rc13

### System information

n/a

### Issue description

Builtins containers with int/float numbers are declared in `extension_api.json`:

```json
{
	"name": "Vector2",
	"indexing_return_type": "float",
	"is_keyed": true,
	"members": [
		{
			"name": "x",
			"type": "float"
		},
		{
			"name": "y",
			"type": "float"
		}
	],
```

However the `float` type here always means 32bits long floating number (so C style float), this is the same for Vector2i which members are declared as int (meaning 32 bits long interger, C int32_t).
This is totally in line with the Vector2/Vector2i classes in Godot C++ code, however it is very misleading in `extension_api.json` given `int` and `float` are defined there depending of the build configuration:

```python
"builtin_class_sizes": [
{
	"build_configuration": "float_32",
	"sizes": [
                [...]
		{
			"name": "int",
			"size": 8
		},
		{
			"name": "float",
			"size": 8
		},
                [...]
		{
			"name": "Vector2",
			"size": 8
		},
		{
			"name": "Vector2i",
			"size": 8
		},
                [...]
},
[...]
{
	"build_configuration": "double_64",
	"sizes": [
                [...]
		{
			"name": "int",
			"size": 8
		},
		{
			"name": "float",
			"size": 8
		},
                [...]
		{
			"name": "Vector2",
			"size": 16
		},
		{
			"name": "Vector2i",
			"size": 8
		},
                [...]
},
```

So:
- int and float are **always** 8bytes long (so int64_t / double in C)
- Vector2i is **always** made of 2 4bytes long integers (so int32_t in C)
- Vector2 is either 2 4bytes (float in C) or 2 8bytes (double in C)

## An attempt to improve this

I think we should introduce a `meta` field in the Vector2/Vector2i declarations to further specify the size of those int/float (this is already what is done in the types used in classes' methods)

```json
{
	"name": "Vector2",
	"indexing_return_type": "float",
	"is_keyed": true,
	"members": [
		{
			"name": "x",
			"type": "float",
			"meta": "double"  // would be "float" if we were in float_32/float_64
		},
		{
			"name": "y",
			"type": "float",
			"meta": "double"
		}
	],
```

PS: I focus on Vector2/Vector2i here, but this issue is also relevant for Vector3/Vector4 (in fact any builtin that declares containing float/int in extension_api.json)

### Steps to reproduce

n/a

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by touilleMan**
fix #64189

(replaces #64365 that was incorrectly based on a branch own by `godotengine/godot` instead of `touilleMan/godot`)

### CODE REVIEW

**Comment by touilleMan** [comment]
@akien-mga rebased

**Comment by reduz** [comment]
I don't entirely understand the need for this, can't this information be inferred from the existing json?

**Comment by touilleMan** [comment]
@reduz see #64189

*tl;dr:*

currently `float` in `gdextension_api.json` means different things (e.g. in `float_32` config, `Vector2` type contains `float` fields that are 4bytes, where the `float` builtin type is 8bytes long).

This makes sens from GDScript point of view (given `foo = 0.1` and `foo = Vector2(0.1, 0).x` both return the same real variant. 

However it is misleading in GDExtension where providing a parameter with the wrong size means memory corruption 😄 

Currently to know that Vector2 is composed of 4bytes long floats, one has to do arithmetic with builtin total size per-field offset information...

On top of that it's easy to jump to the conclusion that "`float` builtin is 4bytes long, builtin fields are 4bytes long on config float_32/64 and 8 bytes on double_32/64"... then missing the fact Color's `float` fields are always 4bytes long !

So this PR provide a `meta` field for the builtins attribute that works exactly like the `meta` field already existing for classes methods' parameters.

**Comment by touilleMan** [comment]
rebased ;-)

**Review by groud — CHANGES_REQUESTED** [review]
We reviewed this PR during the GDextension PR review meeting. The feature seems fine and useful, though we are not sure the "size" field is, as the size itself could be inferred from the type's size provided in the meta field.

The PR should be split into two, one with the "meta" field which was approved, and maybe another with the size if we find use case where that would be useful.

**Comment by touilleMan** [comment]
@groud Updated to remove the `size` field ;-)

**Review by groud — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
